lib/mail/network/retriever_methods/imap.rb

```diff
@@ -153,8 +153,13 @@ def validate_options(options)
       # Start an IMAP session and ensures that it will be closed in any case.
       def start(config=Mail::Configuration.instance, &block)
         raise ArgumentError.new("Mail::Retrievable#imap_start takes a block") unless block_given?
-
-        imap = Net::IMAP.new(settings[:address], settings[:port], settings[:enable_ssl], nil, false)
+        tmp_ssl = settings[:enable_ssl]
+        tmp_ssl = false unless settings[:starttls_options].nil?
```

Comment:
Since these options are mutually exclusive, we can raise an ArgumentError if both are set.